src/components/dialogs/substation-creation-dialog.js

```diff
@@ -0,0 +1,255 @@
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../utils/messages';
+import { createSubstation } from '../../utils/rest-api';
+import { validateField } from '../util/validation-functions';
+import { Autocomplete } from '@material-ui/lab';
+import { useParameterState } from '../parameters';
+import { PARAM_LANGUAGE } from '../../utils/config-params';
+import { getComputedLanguage } from '../../utils/language';
+
+const useStyles = makeStyles((theme) => ({
+    helperText: {
+        margin: 0,
+        marginTop: 4,
+    },
+    popper: {
+        style: {
+            width: 'fit-content',
+        },
+    },
+    h3: {
+        marginBottom: 0,
+        paddingBottom: 1,
+    },
+    h4: {
+        marginBottom: 0,
+    },
+}));
+
+/**
+ * Dialog to create a substation in the network
+ * @param {Boolean} open Is the dialog open ?
+ * @param {EventListener} onClose Event to close the dialog
+ * @param selectedNodeUuid : the currently selected tree node
+ */
+const SubstationCreationDialog = ({ open, onClose, selectedNodeUuid }) => {
+    const classes = useStyles();
+
+    const [languageLocal] = useParameterState(PARAM_LANGUAGE);
+
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const intl = useIntl();
+    const intlRef = useIntlRef();
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [substationId, setSubstationId] = useState('');
+
+    const [substationName, setSubstationName] = useState('');
+
+    const [substationCountry, setSubstationCountry] = useState('');
+
+    const [errors, setErrors] = useState(new Map());
+
+    const handleChangeSubstationId = (event) => {
+        setSubstationId(event.target.value);
+    };
+
+    const handleChangeSubstationName = (event) => {
+        setSubstationName(event.target.value);
+    };
+
+    const handleChangeCountry = (newValue) => {
+        setSubstationCountry(newValue);
+    };
+
+    const handleSave = () => {
+        let tmpErrors = new Map(errors);
+
+        tmpErrors.set(
+            'substation-id',
+            validateField(substationId, {
+                isFieldRequired: true,
+            })
+        );
+
+        setErrors(tmpErrors);
+
+        // Check if error list contains an error
+        let isValid =
+            Array.from(tmpErrors.values()).findIndex((err) => err.error) === -1;
+
+        if (isValid) {
+            createSubstation(
+                studyUuid,
+                selectedNodeUuid,
+                substationId,
+                substationName,
+                substationCountry
+            )
+                .then(() => {
+                    handleCloseAndClear();
+                })
+                .catch((errorMessage) => {
+                    displayErrorMessageWithSnackbar({
+                        errorMessage: errorMessage,
```

Comment:
errorMessage is empty there. We want the error message from the back nope ?

Reply:
fixed the problem was backend